frontend/src/lib/OutputArtifactLoader.ts

```diff
@@ -198,6 +217,275 @@ export class OutputArtifactLoader {
+    let res: GetContextByTypeAndNameResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getContextByTypeAndName(request);
+    } catch (err) {
+      err.message = 'Failed to getContextsByTypeAndName: ' + err.message;
+      throw err;
+    }
+
+    const context = res && res.getContext();
+    if (!context) {
+      throw new Error("getContextByTypeAndName didn't have a context");
+    }
+    return context;
+  }
+
+  public static async getExecutionInContextWithPodName(
+    argoPodName: string,
+    context: Context,
+  ): Promise<Execution> {
+    const contextId = context.getId();
+    if (!contextId) {
+      throw new Error('Context must have an ID');
+    }
+
+    const request = new GetExecutionsByContextRequest();
+    request.setContextId(contextId);
+    let res: GetExecutionsByContextResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getExecutionsByContext(request);
+    } catch (err) {
+      err.message = 'Failed to getExecutionsByContext: ' + err.message;
+      return err;
+    }
+
+    const executionList = (res && res.getExecutionsList()) || [];
+    const foundExecution = executionList.find(execution => {
+      const properties = execution.getPropertiesMap();
+      const executionPodName = properties.get('kfp_pod_name');
+      if (!executionPodName) {
+        return false;
+      }
+      const executionState = properties.get('state');
+      if (!executionState) {
+        return false;
+      }
+      return (
+        executionPodName.getStringValue() === argoPodName &&
+        executionState.getStringValue() === 'complete'
+      );
+    });
+    if (!foundExecution) {
+      logger.verbose("Couldn't find corresponding execution in context");
+      throw new Error("Couldn't find corresponding execution in context");
+    }
+    return foundExecution;
+  }
+
+  public static async getOutputArtifactsInExecution(execution: Execution): Promise<Artifact[]> {
+    const executionId = execution.getId();
+    if (!executionId) {
+      throw new Error('Execution must have an ID');
+    }
+
+    const request = new GetEventsByExecutionIDsRequest();
+    request.addExecutionIds(executionId);
+    let res: GetEventsByExecutionIDsResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getEventsByExecutionIDs(request);
+    } catch (err) {
+      err.message = 'Failed to getExecutionsByExecutionIDs: ' + err.message;
+      return err;
+    }
+
+    const eventsList = (res && res.getEventsList()) || [];
+    const outputArtifactIds: number[] = [];
+    eventsList.forEach(event => {
+      if (event.getType() === Event.Type.OUTPUT) {
+        const artifactId = event.getArtifactId();
+        if (artifactId) {
+          outputArtifactIds.push(artifactId);
+        }
+      }
+    });
+
+    const artifactsRequest = new GetArtifactsByIDRequest();
+    outputArtifactIds.forEach(artifactId => artifactsRequest.addArtifactIds(artifactId));
```

Comment:
```suggestion
    artifactsRequest.setArtifactIdsList(outputArtifactIds);
```
looks like this is equivalent